### PR TITLE
Adding MVP Basics structure

### PR DESCRIPTION
**Pull request by julian-mainardis-globant**
## Description
MVP Basic structure added to the project

## Story Type

- New feature :white_check_mark:
- Change request
- Bug fix
- Tech

## How Can This Be Tested?
Opening the project in the main screen

## Evidences

| First Header  | Second Header |
| ------------- | ------------- |
| image  | image  |

## Tested on

- Nexus 5X
- Nexus S

### CODE REVIEW

File: app/src/main/java/com/mobile/tandil/javabaseproject/activity/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        new ParkingPresenter(new ParkingModel(), new ParkingView(this));
+
+    }
+
```
Comment (SartiMau) [inline app/src/main/java/com/mobile/tandil/javabaseproject/activity/MainActivity.java]: remove this extra space please, never put an empty line before c

File: app/src/main/java/com/mobile/tandil/javabaseproject/activity/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+public class MainActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        new ParkingPresenter(new ParkingModel(), new ParkingView(this));
+
```
Comment (SartiMau) [inline app/src/main/java/com/mobile/tandil/javabaseproject/activity/MainActivity.java]: remove this extra space please, never put an empty line before close curly brackets

File: app/src/main/java/com/mobile/tandil/javabaseproject/activity/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import com.mobile.tandil.javabaseproject.mvp.view.ParkingView;
+
+public class MainActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        new ParkingPresenter(new ParkingModel(), new ParkingView(this));
```
Comment (SartiMau) [inline app/src/main/java/com/mobile/tandil/javabaseproject/activity/MainActivity.java]: this has no sense, please assign this to a variable

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/base/ActivityView.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+    private WeakReference<Activity> activityRef;
+
+    public ActivityView(Activity activity) {
+        activityRef = new WeakReference<>(activity);
+    }
+
+    @Nullable
+    public Activity getActivity() { return activityRef.get(); }
+
```
Comment (DiegoDiazGlobant) [inline app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/base/ActivityView.java:19]: remove extra lines

File: app/src/main/java/com/mobile/tandil/javabaseproject/activity/MainActivity.java (hunk 1 above)
Comment (julian-mainardis-globant) [reply app/src/main/java/com/mobile/tandil/javabaseproject/activity/MainActivity.java]: done

File: app/src/main/java/com/mobile/tandil/javabaseproject/activity/MainActivity.java (hunk 2 above)
Comment (julian-mainardis-globant) [reply app/src/main/java/com/mobile/tandil/javabaseproject/activity/MainActivity.java]: done

File: app/src/main/java/com/mobile/tandil/javabaseproject/activity/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import android.os.Bundle;
+
+import com.mobile.tandil.javabaseproject.R;
+import com.mobile.tandil.javabaseproject.mvp.model.ParkingModel;
+import com.mobile.tandil.javabaseproject.mvp.presenter.ParkingPresenter;
+import com.mobile.tandil.javabaseproject.mvp.view.ParkingView;
+
+public class MainActivity extends AppCompatActivity {
+
+    private ParkingPresenter presenter;
```
Comment (SartiMau) [inline app/src/main/java/com/mobile/tandil/javabaseproject/activity/MainActivity.java]: we should use the more higher type in the hierarchy, so please change it to Parking Contract.Presenter

**Review by SartiMau — CHANGES_REQUESTED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by mariano98montero — APPROVED** [review]

**Review by DiegoDiazGlobant — APPROVED** [review]

**Pull request merged**